src/constitutive_relations/column_integrators/EvaluatorColumnIntegrator.hh

```diff
@@ -130,20 +131,63 @@ EvaluatorColumnIntegrator<Parser, Integrator>::Evaluate_(
   for (int col = 0; col != res.MyLength(); ++col) {
     // for each column, loop over cells calling the integrator until stop is
     // requested or the column is complete
-    AmanziGeometry::Point val(0., 0.);
+    AmanziGeometry::Point val(0., 0., NAN);
     auto col_cell = mesh->columns.getCells(col);
-    for (int i = 0; i != col_cell.size(); ++i) {
-      bool completed = integrator.scan(col, col_cell[i], val);
-      if (completed) break;
+    double h_top = mesh->getCellCentroid(col_cell[0])[2]
+        + mesh->getCellVolume(col_cell[0]) * integrator.coefficient(col) / 2;
+    double h_bot = mesh->getCellCentroid(col_cell[col_cell.size() - 1])[2]
+        - mesh->getCellVolume(col_cell[col_cell.size() - 1]) * integrator.coefficient(col) / 2;
+    double h_end, h_half0, h_half1;
+
+    if (plist_.get<std::string>("evaluator type") == "water table depth") {
```

Comment:
No, let's talk about the structure of this.  EvaluatorColumnIntegrator shouldn't have water-table-specific code here.  Rather, you should create a new class (which maybe inherits from EvaluatorColumnIntegrator) and implement it there.